csm_test_utils/rds/rds_backup.py

```diff
@@ -0,0 +1,132 @@
+    response = requests.get(url = url, params = request_params, headers = request_headers)
+    return response
+
+
+def get_duration(start_time: str, end_time: str):
+    time_delta = (format_date_time(end_time) - format_date_time(start_time)).total_seconds()
+    return time_delta
+
+
+def format_date_time(date_time: str):
+    return datetime.strptime(date_time, "%Y-%m-%dT%H:%M:%S%z")
+
+
+def get_rds_backup_status(token: str, project_id: str, instance_id: str, backup_type: str) -> str:
+    request_params = {'instance_id': instance_id, 'backup_type': backup_type}
+    response = get_rds_backup_info(token, project_id, **request_params)
+    return response
+
+
+def report(client: Client, token: str, project_id: str, **request_params):
+    """Send request and write metrics to telegraf"""
+    collection = MetricCollection()
+    try:
+        influx_row = Metric(RDS_BACKUP)
+        target_req = get_rds_backup_info(token, project_id, **request_params)
+        if target_req.ok:
+            backups = target_req.json()["backups"]
+            for backup in backups:
+                influx_row.add_tag("id_backup", backup["id"])
+                influx_row.add_tag("status", backup["status"])
+                influx_row.add_tag("size", backup["size"])
+                influx_row.add_value("backup_duration", get_duration(backup["begin_time"], backup["end_time"]))
+                collection.append(influx_row)
+        else:
+            influx_row.add_tag("status", "request_failed")
+            influx_row.add_tag("host", "scn6")
+            influx_row.add_tag("reason", "fail")
+            influx_row.add_value("elapsed", target_req.elapsed.seconds)
+            collection.append(influx_row)
+    except (IOError, HTTPError) as Error:
+        influx_row = Metric(CSM_EXCEPTION)
+        influx_row.add_tag("Reporter", RDS_BACKUP)
+        influx_row.add_tag("Status", "RDS Unavailable")
+        influx_row.add_value("Value", Error)
+        collection.append(influx_row)
+    except Exception as Ex:
+        return LOGGER.exception(Ex)
+    client.report_metric(collection)
+
+
+AGP = sub_parsers.add_parser(RDS_BACKUP, add_help=False, parents=[base_parser])
+AGP.add_argument("--instance_id", help = "RDS instance ID")
+AGP.add_argument("--cloud_config", help = "Clouds config file")
+AGP.add_argument("--cloud_name", help = "Name of cloud")
+
+
+def main():
+    args, _ = AGP.parse_known_args()
+    token, project_id = get_auth_token(args.cloud_config, args.cloud_name)
+    request_params = {'instance_id': args.instance_id, 'backup_type': 'auto'}
+    client = Client(args.target, args.telegraf)
+    setup_logger(LOGGER, "rds_backup_monitor", log_dir = args.log_dir, log_format = "[%(asctime)s] %(message)s")
+    LOGGER.info(f"Started monitoring of {client.url} (telegraf at {client.tgf_address})")
+    while True:
+        try:
+            report(client, token, project_id, **request_params)
+            time.sleep(60)
```

Comment:
once per minute is too often, once per hour on more is ok

Reply:
Done